### DISCUSSION TITLE
Units Units Units

### DISCUSSION THREAD

**Discussion by alexanderson1993** [Thorium-Sim/thorium-nova#348] [Q&A]
I was doing some research in to the power generation and distribution, and I came to realize, once again, that we are embarking on some pretty deep, pretty complicated stuff.

In Thorium Classic, power was represented by units. What's a unit? Don't know.

Thorium Nova tries to increase immersion by using realistic distances, sizes, and units for everything. So if we're representing power and energy, we should use watts and watt-hours, right? And we'll have batteries, which commonly use amps to represent their capacity, which means the ship needs a voltage of some kind to know how much power the batteries actually have stored.

Naturally, we'll have to do some hand-woven space magic to make any of this work, since the power needed for things like warp engines, transporters, or shields are astronomically high. But it might be worth handwaving some of this other stuff too.

Except then you realize that this affects everything on the ship, from end to end. Fuel will be consumed to generate the power on the ship, and after some research I found that the fusion of 1 gram of deuterium  produces 226 megawatt-hours of energy at 100% efficiency. So now our inventory items need to have a unit for how much energy they contain?

And what about weapons? A 1 megaton TNT explosion releases 1.16 terawatt-hours, and the shields need to deflect the portion of that blast directed towards the ship. Let's assume that it's just 30% of the explosion, just for fun. So 1.16 TWh * 0.3 / 226 MWh = 1.5Kg of fuel needed to deflect that explosion.  Also, I imagine future space weapons will be much more powerful. 

Speaking of weapons, it would seem Star Trek phasers are in the 200 terawatt range. A 5 second burst would require 1.2 kilograms of fuel. If it were stored in super-cold conditions, that fuel would fill 8 gallon jugs, or 30 liters. 

I guess this is a question of how far is too far, or if it does make sense to connect all the units like this. At some level, there will have to be some mechanism that converts weapons impacts into shield discharge, and the shield charge comes from the power, and power comes from fuel. So using common units for all of those does make sense. But is it worth it to hand wave some of it and fudge the numbers?

But there is also a question of how much of this to expose to the crew, or whether the controls should be kept as simple as possible. 

Also, vastly different power requirements among systems might make the while power distribution system a joke - you pretty much just put power in the engines, weapons, and shields; whatever power is left over will be more than enough for all of the other systems. That makes me want to fudge how much power systems consume to do what they do to make power distribution more interesting. Then again, most bridge simulators only let you distribute power to engines, weapons, and shields, so maybe that's good enough.

All of this also means that there will be a delicate balance when creating content for Thorium Nova. If you can configure the yield of torpedoes, then you might decide that one faction gets special torpedos that have a slightly higher yield, and now all of a sudden everyone else runs out of fuel faster because they keep needing to recharge their shields.

Honestly, I don't know how it will all work out, so any feedback or suggestions are welcome.

**Comment by 1deejay** [comment]
First off, the equation is very easy. Amps X Volts = Watts

Batteries telling you the capacity in ampere hours is incredibly frustrating because if you change the voltage it changes the entire equation. Fortunately, batteries are more and more using Watt Hours (Wh).

Most bridge sims letting you distribute power to just a few places is part of keeping things simple enough for players to be able to get things done in a half our or so time limit, on top of only going up to four or so players. This being a more full sim, we can get more complicated.

**Comment by alexanderson1993** [comment]
Throwing this in here so I don't lose track of it

17 MeV per deuterium/tritium reaction, 2.99x10^23 atoms per gram of deuterium, 226.2 MWh
5.95 MeV per deuterium reaction, 2.99x10^23 atoms per gram of deuterium, 79.18 MWh
1.44 MeV per hydrogen reaction, 5.974x10^23 atoms per gram of hydrogen, 38.29 MWh
